src/adaptix/_internal/morphing/constant_length_tuple_provider.py

```diff
@@ -0,0 +1,326 @@
+import collections.abc
+import typing
+from typing import Collection, Mapping, Tuple
+
+from ..common import Dumper, Loader
+from ..compat import CompatExceptionGroup
+from ..definitions import DebugTrail
+from ..feature_requirement import HAS_UNPACK
+from ..load_error import (
+    AggregateLoadError,
+    ExcludedTypeLoadError,
+    ExtraItemsError,
+    LoadError,
+    NoRequiredItemsError,
+    TypeLoadError,
+)
+from ..provider.essential import CannotProvide, Mediator
+from ..provider.provider_template import for_predicate
+from ..provider.request_cls import (
+    DebugTrailRequest,
+    DumperRequest,
+    LoaderRequest,
+    LocMap,
+    StrictCoercionRequest,
+    TypeHintLoc,
+    get_type_from_request,
+    try_normalize_type,
+)
+from ..struct_trail import append_trail, render_trail_as_note
+from .provider_template import DumperProvider, LoaderProvider
+
+CollectionsMapping = collections.abc.Mapping
+
+
+@for_predicate(Tuple)
+class ConstantLengthTupleProvider(LoaderProvider, DumperProvider):
+    def _provide_loader(self, mediator: Mediator, request: LoaderRequest) -> Loader:
+        norm = try_normalize_type(get_type_from_request(request))
+        if len(norm.args) > 1 and norm.args[1] == Ellipsis:
+            raise CannotProvide
+        if HAS_UNPACK and any(arg.origin == typing.Unpack for arg in norm.args if arg != Ellipsis):
+            raise CannotProvide(
+                "typing.Unpack isn't supported yet",
+                is_terminal=True,
+                is_demonstrative=True
+            )
+
+        loaders = mediator.mandatory_provide_by_iterable(
+            [
+                LoaderRequest(
+                    loc_map=LocMap(
+                        TypeHintLoc(type=tp.source)
+                    )
```

Comment:
Each type is attached to a specific generic argument, so each request loc_map must contain ``GenericParamLoc(generic_pos=i)``